src/test/ui/nll/issue-52534-1.stderr

```diff
@@ -0,0 +1,59 @@
+error[E0597]: `x` does not live long enough
+  --> $DIR/issue-52534-1.rs:19:9
+   |
+LL |     fn bar(&self, x: &u32) -> &u32 {
+   |            -----              ---- has type `&'0 u32`
+   |            |
+   |            has type `&'0 Test`
+LL |         let x = 22;
+LL |         &x
+   |         ^^ `x` would have to be valid for `'0`
+LL |     }
+   |     - ...but `x` is only valid for the duration of the `bar` function, so it is dropped here while still borrowed
```

Comment:
I'm wondering wether it might make sense to add a `note` for these cases with text along the lines of "you cannot return a borrow to a struct instance created within a scope, you can only return borrows to things passed in through an argument" with a link to https://doc.rust-lang.org/book/second-edition/ch04-02-references-and-borrowing.html#dangling-references...

Reply:
That's a good idea. I've tweaked the text slightly and added a note with this.